### ISSUE TITLE
Add more crd annotations to cover subresources and additional printer columns.

### ISSUE THREAD

**Issue by iocanel** [fabric8io/kubernetes-client#2736]
We currently have a nice set of annotations for defining core crd configuration.

External tooling could benefit form having access to additional annotations for specifying subresources and additional printer columns.
Even though these annotations, won't be used directly from the client (yet). It does make sense to propose them for downstream projects to use.

Dekorate does provide those and we could directly port them here: https://github.com/dekorateio/dekorate/tree/master/annotations/crd-annotations/src/main/java/io/dekorate/crd/annotation

The annotations of interest are:

- Status
- PrinterColumn
- SpecReplicas
- StatusReplicas
- LabelSelector

**Comment by metacosm** [comment]
We should also have a `ShortName` annotation, no? Another question is also whether it makes sense to group some of these values into a single annotation instead of multiplying them? @iocanel

**Comment by metacosm** [comment]
This has already been done. However, we should move the CRD-related annotations to the `io.fabric8.crd.annotation` package in the `crd-generator` module as we now have a mix of annotations in this module and some other in `io.fabric8.kubernetes.model.annotation`, with different retention policies.

**Comment by metacosm** [comment]
Still waiting on your opinion, @iocanel @manusa @rohanKanojia…

**Comment by rohanKanojia** [comment]
Hi, Sorry for the late reply. I'm sorry but I'm not able to clearly understand what you're asking/suggesting. Do we want to merge both annotations in one place? Is our current approach of using annotations being used at runtime limiting us for some use cases in crd-generator? Maybe we can create a separate issue for discussing them in detail?

**Comment by manusa** [comment]
I see many drawbacks when doing things at build-time, because it always implies a recompilation of your project which in some restricted corporate environments can be very hard.

The `crd-generator` is now intended to be used at build-time since it relies on annotations and APT, but we are already discussing that it makes sense to have a public API to expose the logic of the generator (#2836). I really see this as the way forward since I feel that the YAML generation is somewhat unnecessary unless you want interoperability amongst clients. "Imagine a world with no YAML, it isn't hard to do, Nothing to kill or die for...."

So basically in an ideal YAML-less world, you could use the client in a Groovy/JBang/JShell/... script to deploy your resources with no need to code a single line in YAML. For this kind of dynamic use-case (replace Helm for a real script) APT wouldn't really apply.

Regarding annotations, we already have many in the `io.fabric8.kubernetes.model.annotation` of the `kubernetes-model-common` module. I don't see the point of moving them elsewhere. Why not add the rest of them to this package?

> or the user would need to override the appropriate methods

I think that this was one of the initial alternatives when we started to deal with the annotations for CRs but I think it was agreed to use annotations instead (but I can't recall very well).

**Comment by metacosm** [comment]
I don't think that a YAML-less world is realistic, unfortunately. 
Am I right in understanding that the consensus is that the annotations should be moved all to the `io.fabric8.kubernetes.model.annotation` package then? Should we change the retention policy as well so that they all match?
